### PR TITLE
Drop JSHint and use ESLint

### PR DESCRIPTION
**Pull request by iivic**
LEARNER-1167

We are replacing JSHint with ESLint entirely.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @iivic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=1303&repo=edx%2Fecommerce"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by iivic** [comment]
@clintonb please take a look at the change.
After running `make validate_js` we will get `2970 problems (2961 errors, 9 warnings)`.
 I tried https://github.com/TheSavior/ESLint-Formatter to format automatically in sublime but it refuses to format the code.
Can you suggest a solution or at least a different approach?

**Comment by clintonb** [comment]
http://eslint.org/docs/user-guide/command-line-interface

Use `--fix`.

**Comment by iivic** [comment]
@clintonb please review and maybe propose another reviewer?

File: package.json [hunk 1]
```diff
@@ -10,13 +10,14 @@
     "requirejs": "2.3.3"
   },
   "devDependencies": {
+    "eslint": "^3.19.0",
```
Comment (clintonb) [inline package.json]: Pin all requirements to a specific release, rather than a range, to avoid issues with dependencies changing unexpetedly.

**Review by clintonb — APPROVED** [review]

**Review by AlasdairSwan — APPROVED** [review]
👍

**Pull request merged**